app/controllers/sales_logs_controller.rb

```diff
@@ -40,6 +40,28 @@ def edit
     end
   end
 
+  def destroy
+    @log = SalesLog.visible.find_by(id: params[:id])
+
+    render_not_found and return unless @log
```

Comment:
to prevent from littering every controller method with this line you can use `rescue_from` combined with `ActiveRecord::RecordNotFound` to handle these at a higher level